pkg/fleet/installer/setup/common/config_test.go

```diff
@@ -14,48 +11,153 @@ import (
-	assertFileContent(t, datadogConfFile, `api_key: a
-key: value
-logs_enabled: true
-tags:
-- k1:v1
-- k2:v2
-`)
-
-	assertFileContent(t, logsConfFile, `logs:
-- type: file
-  path: /var/log/app.log
-  service: app
-`)
-	assertFileContent(t, sparkConfigFile, `instances:
-- spark_url: http://localhost:8080
-  cluster_name: cluster
-`)
+func TestMergeConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	oldConfig := `---
+api_key: "0987654321"
+hostname: "old_hostname"
+env: "old_env"
+`
+	err := os.WriteFile(filepath.Join(tempDir, datadogConfFile), []byte(oldConfig), 0644)
+	assert.NoError(t, err)
+	config := Config{}
+	config.DatadogYAML.APIKey = "1234567890" // Required field
+	config.DatadogYAML.Hostname = "new_hostname"
+
+	err = writeConfigs(config, tempDir)
+	assert.NoError(t, err)
+
+	// Check datadog.yaml
+	datadogConfigPath := filepath.Join(tempDir, datadogConfFile)
+	datadogYAML, err := os.ReadFile(datadogConfigPath)
+	assert.NoError(t, err)
+	var datadog map[string]interface{}
+	err = yaml.Unmarshal(datadogYAML, &datadog)
+	assert.NoError(t, err)
+	assert.Equal(t, map[string]interface{}{
+		"api_key":  "1234567890",
+		"hostname": "new_hostname",
+		"env":      "old_env",
+	}, datadog)
+}
+
+func TestInjectTracerConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	config := Config{}
+	config.InjectTracerYAML = InjectTracerConfig{
+		Version:       1,
+		ConfigSources: "env",
+		AdditionalEnvironmentVariables: []InjectTracerConfigEnvVar{
+			{
+				Key:   "DD_ENV",
+				Value: "prod",
+			},
+		},
+	}
+
+	err := writeConfigs(config, tempDir)
+	assert.NoError(t, err)
+
+	// Check inject/tracer.yaml
+	injectTracerConfigPath := filepath.Join(tempDir, injectTracerConfigFile)
+	assert.FileExists(t, injectTracerConfigPath)
+	injectTracerYAML, err := os.ReadFile(injectTracerConfigPath)
+	assert.NoError(t, err)
+	var injectTracer map[string]interface{}
+	err = yaml.Unmarshal(injectTracerYAML, &injectTracer)
+	assert.NoError(t, err)
+	assert.Equal(t, map[string]interface{}{
+		"version":        1,
+		"config_sources": "env",
+		"additional_environment_variables": []interface{}{
+			map[string]interface{}{
+				"key":   "DD_ENV",
+				"value": "prod",
+			},
+		},
+	}, injectTracer)
+}
+
+func TestIntegrationConfigInstanceSpark(t *testing.T) {
+	tempDir := t.TempDir()
+	config := Config{
+		IntegrationConfigs: make(map[string]IntegrationConfig),
+	}
+	config.IntegrationConfigs["spark.d/kebabricks.yaml"] = IntegrationConfig{
+		Logs: []IntegrationConfigLogs{
+			{
+				Type:    "file",
+				Path:    "/databricks/spark/work/*/*/stderr",
+				Source:  "worker_stderr",
+				Service: "databricks",
+			},
+		},
+		Instances: []any{
+			IntegrationConfigInstanceSpark{
+				SparkURL:         "http://localhost:4040",
+				SparkClusterMode: "spark_driver_mode",
+				ClusterName:      "big-kebab-data",
+				StreamingMetrics: true,
+			},
+		},
+	}
+
+	err := writeConfigs(config, tempDir)
+	assert.NoError(t, err)
+
+	// Check spark.d/kebabricks.yaml
```

Comment:
a little verification that agent config has logs enabled here too would be helpful